saliency/metrics/pic.py

```diff
@@ -0,0 +1,377 @@
+  pixel_mask = pixel_mask.copy()
+  height = pixel_mask.shape[0]
+  width = pixel_mask.shape[1]
+  pixel_mask[
+    [0, 0, height - 1, height - 1], [0, width - 1, 0, width - 1]] = True
+
+  mean_color = np.mean(full_img, axis=(0, 1))
+
+  # If the mask consists of all pixels set to True then return the original
+  # image.
+  if np.all(pixel_mask):
+    return full_img
+
+  blurred_img = full_img * np.expand_dims(pixel_mask, axis=2).astype(
+      np.float32)
+
+  # Interpolate the unmasked values of the image pixels.
+  for channel in range(channels):
+    data_points = np.argwhere(pixel_mask > 0)
+    data_values = full_img[:, :, channel][tuple(data_points.T)]
+    unknown_points = np.argwhere(pixel_mask == 0)
+    interpolated_values = interpolate.griddata(np.array(data_points),
+                                               np.array(data_values),
+                                               np.array(unknown_points),
+                                               method=method,
+                                               fill_value=mean_color[channel])
+    blurred_img[:, :, channel][tuple(unknown_points.T)] = interpolated_values
+
+  if not has_color_channel:
+    blurred_img = blurred_img[:, :, 0]
+
+  if issubclass(data_type.type, np.integer):
+    blurred_img = np.round(blurred_img)
+
+  return blurred_img.astype(data_type)
+
+
+def generate_random_mask(image_shape: Sequence, fraction=0.01) -> np.ndarray:
+  """Generates a random pixel mask.
+
+    The result of this method can be used as the initial seed to generate
+    a 'fully' blurred image with the help of the create_blurred_image(...)
+    method.
+
+    Args:
+      image_shape: shape of the image for which the mask should be generated.
+        Only the first two dimensions are uses.
+      fraction: the fraction of the mask pixels that should be set to true.
+        The valid value range is [0.0, 1.0]. Set the value to 0.0 if no
+        information from the original image should be used in the blurred image.
+    Returns:
+      The binary mask with the `fraction` of elements set to True.
+  """
+  height = image_shape[0]
+  width = image_shape[1]
+  mask = np.zeros(shape=[height, width], dtype=bool)
+  size = mask.size
+  indices = np.random.choice(size, replace=False, size=int(size * fraction))
+  mask[np.unravel_index(indices, mask.shape)] = True
+  return mask
+
+
+def estimate_image_entropy(image: np.ndarray) -> float:
+  """Estimates the amount of information in a given image.
+
+    Args:
+      image: an image, which entropy should be estimated. The dimensions of the
+        array should be [H, W, C] of type uint8.
+    Returns:
+      The estimated amount of information in the image.
+  """
+  buffer = io.BytesIO()
+  pil_image = Image.fromarray(image)
+  pil_image.save(buffer, format='webp', lossless=True, quality=100)
+  buffer.seek(0, os.SEEK_END)
+  length = buffer.tell()
+  buffer.close()
+  return length
+
+
+class ComputePicMetricResult(NamedTuple):
```

Comment:
consider renaming from "Compute". Since "Compute" is a verb, it's confusing that it is just a container for already computed results.

Reply:
I see what you mean. I guess I'd maybe use just "PicMetricResult", but I see how that's no longer parallel to ComputePicMetricError. I think it's fine as is, but will leave it up to you to decide/resolve.

Reply:
Renaming makes sense. Done.